File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        Map<Integer, Integer> cnt = new HashMap<>();
        Set<Integer> seen = new HashSet<>();

        for (int num: arr) {
            cnt.put(num, cnt.getOrDefault(num, 0) + 1);
        }

        for (int num: cnt.values()) {
            if (seen.contains(num)) return false;
            seen.add(num);
        }

        return true;
    }
}

File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        unordered_map<int, int> cnt;
        set<int> seen;

        for (int num: arr) {
            cnt[num]++;
        }

        for (auto elem: cnt) {
            if (seen.find(elem.second) != seen.end()) {
                return false;
            }
            seen.insert(elem.second);
        }

        return true;
    }
};